File: Code/Components/DanpheEMR.ServerModel/WardSupplyModels/WARDDispatchItemsModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class WARDDispatchItemsModel
    {
        [Key]
        public int DispatchItemId { get; set; }

        public int DispatchId { get; set; }

        public int? RequisitionItemId { get; set; }

        public int ItemId { get; set; }

        public string ItemName { get; set; }

        public string BatchNo { get; set; }

        public DateTime ExpiryDate { get; set; }

        public int Quantity { get; set; }

        public decimal? MRP { get; set; }

        public decimal SubTotal { get; set; }

        public string Remark { get; set; }

        public int CreatedBy { get; set; }

        public DateTime? CreatedOn { get; set; }

        [NotMapped]
        public int TotalQty { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/wardsupply/shared/ward-dispatch-items.model.ts
import {
    NgForm,
    FormGroup,
    FormControl,
    Validators,
    FormBuilder
} from '@angular/forms'
import * as moment from 'moment/moment';

export class WardDispatchItemsModel {
    public DispatchItemId: number = 0;
    public DispatchId: number = 0;
    public RequisitionItemId: number = 0;
    public ItemId: number = 0;
    public ItemName: string = '';
    public BatchNo: string = '';
    public ExpiryDate: string = '';
    public Quantity: number = 0;
    public MRP: number = 0;
    public SubTotal: number = 0;
    public Remark: string = '';
    public CreatedBy: number = 0;
    public CreatedOn: string = moment(new Date()).format('YYYY-MM-DD hh:mm:ss');

    public TotalQty: number = null;
    public DispatchedQty: number = null;
    public enableItmSearch: boolean = true;
    public notValid: boolean = false;
    public IsDuplicateItem: boolean = false;

    constructor() {
    }
}
